varinfo/cmr_search.py

```diff
@@ -36,7 +37,7 @@ def get_granules(
     provider: str | None = None,
     cmr_env: CmrEnvType = CMR_OPS,
     auth_header: str | None = None,
-) -> list:
+) -> Sequence:
```

Comment:
This is a sneaky extra change I'm getting in here due to updating `python-cmr` to a version with type hints.

I'm not a big fan of using `Sequence` here, because I think it really is a `list`, and that's more specific, but the underlying type hint for the return type from `cmr.GranuleQuery.get` is a `Sequence`, so I changed our type hint to be compatible with that.

(The other alternative was to explicitly cast the output of `cmr.GranuleQuery.get` as a `list` - I could be convinced to go either way)